### DISCUSSION TITLE
v17 preliminary support #804

### DISCUSSION THREAD

**Discussion by kvaikla** [cybertec-postgresql/pgwatch2#833] [Q&A] [answered]
Hi, 

Looking for some instruction, how to get my pgwatch2 to monitor pg17.
My setup is "Custom installation", pgwatch2 pre-built packages (rpm)

br
Kaido

**Comment by pashagolub** [comment]
Hello. pgwatch2 is in maintenance mode and will not support PostgreSQL 17. Please, take a look on https://github.com/cybertec-postgresql/pgwatch/

**Comment by kvaikla** [comment]
Clear. But is it this new pgwatch stable and what about documentation like it was https://pgwatch2.readthedocs.io/en/stable/index.html ?
Docker is not an option.
br
Kaido

**Reply by pashagolub** [reply]
The new version is in beta at the moment. Looking forward to official release. Documentation is available at https://pgwat.ch 

Docker is, of course, an option. Please find tutorial here https://pgwat.ch/tutorial/docker_installation.html

**Reply by pashagolub** [reply]
Another option would be to pick up new metrics (if needed) and run pgwatch2 against PostgreSQL 17. At the moment there are no known incompatibilities

**Reply by kvaikla** [reply]
Actually i made extra "Preset metrics config" which is  extract of almoust "full" minus  bgwriter and stat_statements_no_query_text. If it's possible, pls give me acurate selects for bgwriter ja stat_statements_no_query_text for pg17(ee_full_17).
My regular set for pg (ee_full):
`{"kpi": 120, "wal": 60, "locks": 60, "db_size": 300, "archiver": 60, "backends": 60, "bgwriter": 60, "cpu_load": 60, "db_stats": 60, "settings": 7200, "stat_ssl": 120, "wal_size": 120, "locks_mode": 60, "psutil_cpu": 120, "psutil_mem": 120, "index_stats": 900, "instance_up": 60, "psutil_disk": 120, "replication": 120, "sproc_stats": 180, "table_stats": 300, "wal_receiver": 120, "change_events": 300, "stat_activity": 30, "table_io_stats": 600, "replication_slots": 120, "psutil_disk_io_total": 120, "stat_statements_calls": 60, "server_log_event_counts": 60, "stat_statements_no_query_text": 180}`

My set for pg17 (ee_full_17):
`{"kpi": 120, "wal": 60, "locks": 60, "db_size": 300, "archiver": 60, "backends": 60, "cpu_load": 60, "db_stats": 60, "settings": 7200, "stat_ssl": 120, "wal_size": 120, "locks_mode": 60, "psutil_cpu": 120, "psutil_mem": 120, "index_stats": 900, "instance_up": 60, "psutil_disk": 120, "replication": 120, "sproc_stats": 180, "table_stats": 300, "wal_receiver": 120, "change_events": 300, "stat_activity": 30, "table_io_stats": 600, "replication_slots": 120, "psutil_disk_io_total": 120, "stat_statements_calls": 60, "server_log_event_counts": 60}`

**Reply by pashagolub** [reply, marked as answer]
Sure! Please find all up to date metric SQLs here: https://github.com/cybertec-postgresql/pgwatch/blob/master/internal/metrics/metrics.yaml
